xtext/org.icyphy.linguafranca.tests/src/org/icyphy/tests/LinguaFrancaValidationTest.xtend

```diff
@@ -787,14 +800,186 @@ class LinguaFrancaValidationTest {
+        }
+    
+    /**
+     * Maps a type to a list of known bad values.
+     */
+    val primitiveTypeToKnownBad = #{
+            PrimitiveType.BOOLEAN -> #["1 sec", "foo", "\"foo\"", "\'bar\'", "[1]", "{baz: 42}"],
+            PrimitiveType.INTEGER -> #["foo", "\"bar\"", "'baz'", "1 sec", "[1, 2]", "{foo: \"bar\"}"],
+            PrimitiveType.NON_NEGATIVE_INTEGER -> #["-42", "foo", "\"bar\"", "'baz'", "1 sec", "[1, 2]", "{foo: \"bar\"}"],
+            PrimitiveType.TIME_VALUE -> #["foo", "\"bar\"", "'baz'", "\"3 sec\"", "'4 weeks'", "[1, 2]", "{foo: \"bar\"}"],
+            PrimitiveType.STRING -> #["1 msec", "[1, 2]", "{foo: \"bar\"}"]
+        }
+    
+    /**
+     * Maps a type to a list, each entry of which represents a list with 
+     * three entries: a known wrong value, the suffix to add to the reported
+     * name, and the type that it should be.
+     */
+    val compositeTypeToKnownBad = #{
+        UnionType.STRING_OR_STRING_ARRAY -> #[
+            #["[1 msec]", "[0]", PrimitiveType.STRING],
+            #["[foo, {bar: baz}]", "[1]", PrimitiveType.STRING],
+            #["{bar: baz}", "", UnionType.STRING_OR_STRING_ARRAY]
+        ],
+        UnionType.FILE_OR_FILE_ARRAY -> #[
+            #["[1 msec]", "[0]", PrimitiveType.FILE],
+            #["[foo, {bar: baz}]", "[1]", PrimitiveType.FILE],
+            #["{bar: baz}", "", UnionType.FILE_OR_FILE_ARRAY]
+        ]
+    }
+    
+    /**
+     * Synthesize a target property that conforms to the given type, or return
+     * an empty set if it is too complicated.
+     * 
+     * Cases not covered by this function (most composite types) should be added
+     * to the compositeTypeToKnownBad map.
+     *  
+     **/
+    def List<String> synthesizeExamples(TargetPropertyType type, boolean correct) {
+        val values = correct ? primitiveTypeToKnownGood : primitiveTypeToKnownBad 
+        if (type instanceof PrimitiveType) {
+            val examples = values.get(type).toList
+            assertNotNull(examples)
+            return examples
+        } else {
+            if (type instanceof UnionType) {
+                val examples = newLinkedList
+                if (correct) {
+                    type.options.forEach[
+                    if (it instanceof TargetPropertyType) {
+                        synthesizeExamples(it, correct).forEach[examples.add(it)]
+                    } else {
+                        // It must be a plain Enum<?>
+                        examples.add(it.toString())
+                    }
+                ]
+                } else {
+                    // Return some obviously incorrect examples for the common
+                    // case where the options are from an ordinary Enum<?>.
+                    if (!type.options.exists[it instanceof TargetPropertyType]) {
+                        return #["foo", "\"bar\"", "'baz'", "1", "-1", "{x: 42}", "[1, 2, 3]"]
```

Comment:
So we are not synthesizing bad examples here? I see [line 864](https://github.com/icyphy/lingua-franca/pull/272/files#diff-523829f2c3f67a249fddb21468b49d2838ea7e3df75e6ac34f56448aca8f3c05R864) calls `synthesizeExample`.

Reply:
Maybe I don't understand the function yet.. What is the intuition behind the recursion?

Reply:
I'll clean this up a bit and write some more comments. This stuff is a bit idiosyncratic, but it's main purpose is to rely as little as possible on manually provided test vectors. The recursive calls are to deal with composite types (arrays, dictionaries).